### PR TITLE
Restore `clean-conversation-sidebar` feature

### LINKED ISSUE #4238: `clean-conversation-sidebar` is broken
Extension enabled:
Screenshot_20210414_080246
Extension disabled:
Screenshot_20210414_080311

Extension version: 21.4.13
FF version: 78.9.0esr

Note: Sorry if I am missing something here :sweat_smile:

### LINKED ISSUE #4258: `clean-conversation-sidebar` broken
CHROME Version 90.0.4430.72 (Official Build) (64-bit)

Installed the Chrome Extension and it is not working. I am getting the below console error. (It is a private repo). 

Refined GitHub → clean-conversation-sidebar → TypeError: Cannot read property 'remove' of undefined
    at clean (refined-github.js:6568)
    at runFeature (refined-github.js:1942)
    at setupPageLoad (refined-github.js:1955)
    at refined-github.js:1988 
Search issue: https://github.com/sindresorhus/refined-github/issues?q=is%3Aissue+Cannot%20read%20property%20'remove'%20of%20undefined

Open an issue: https://github.com/sindresorhus/refined-github/issues/new?labels=bug&template=bug_report.md&title=%60clean-conversation-sidebar%60%3A%20Cannot%20read%20property%20'remove'%20of%20undefined
		
logError @ refined-github.js:1923

image

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #4238 
Fixes #4258 

## Test URLs

* this PR
* [merged PR](https://github.com/sindresorhus/refined-github/pull/3951)
* [open issue with one linked PR](https://github.com/sindresorhus/refined-github/issues/4397)
* [open issue with a milestone and assignee](https://github.com/microsoft/TypeScript/issues/43501)
* User has triage access
  * [Issue](https://github.com/download-directory/download-directory.github.io/issues/39)
  * [PR](https://github.com/download-directory/download-directory.github.io/pull/37)

## Screenshots

<details>
<summary>This PR</summary>

image
</details>

<details>
<summary>Merged PR</summary>

image
</details>

<details>
<summary>Open issue with linked PR</summary>

image
</details>

<details>
<summary>Open issue with milestone and assignee</summary>

image
</details>

<details>
<summary>PR with no labels (<i>user has triage access</i>)</summary>

image
</details>

<details>
<summary>Issue with labels (<i>user has triage access</i>)</summary>

image
</details>

### CODE REVIEW

File: source/features/clean-conversation-sidebar.tsx [hunk 1]
```diff
@@ -93,9 +93,12 @@ async function clean(): Promise<void> {
...
 	// Labels
-	if (!cleanSection('.sidebar-labels') && !canEditSidebar()) {
-		// Hide header in any case except `canEditSidebar`
-		select('.sidebar-labels div.discussion-sidebar-heading')!.remove();
+	if (!cleanSection('.js-issue-labels') && !canEditSidebar()) {
+		// Hide heading in any case except `canEditSidebar`
+		select('.js-issue-labels')!
+			.closest('.discussion-sidebar-item')!
+			.querySelector(':scope > .discussion-sidebar-heading')!
+			.remove();
```
Comment (fregante) [inline source/features/clean-conversation-sidebar.tsx:101]:
```suggestion
		select('[data-hotkey="l"]')!.remove();
```

Comment (cheap-glitch) [reply source/features/clean-conversation-sidebar.tsx]: Tested this change: the hotkeys are only there if the user has admin privileges in the repo, so we can't fully rely on that selector.

Comment (fregante) [reply source/features/clean-conversation-sidebar.tsx]: It might be worth leaving a comment, I think we talked about this before

Comment (cheap-glitch) [reply source/features/clean-conversation-sidebar.tsx]: Should we still use this selector and keep the "chain" as a fallback? Or just use the chain regardless?

**Comment by yakov116** [comment]
@cheap-glitch you need to take in account users with triage access only

This is what I see on https://github.com/download-directory/download-directory.github.io/pull/37

image

**Comment by yakov116** [comment]
This PR

image

**Comment by cheap-glitch** [comment]
@yakov116 what is wrong in your first screenshot? Is it that the "Labels" section should not be hidden?

Also can you post the DOM of `#partial-discussion-sidebar` on this page with RGH disabled? Thank you

**Comment by yakov116** [comment]
@fregante can you add @cheap-glitch to download-directory so he can test?

File: source/features/clean-conversation-sidebar.tsx (hunk 1 above)
Comment (fregante) [reply source/features/clean-conversation-sidebar.tsx]: Shorter selectors make sense if they shorten the code. We don't duplicate code for negligible speed improvements.

**Comment by cheap-glitch** [comment]
Everything should work properly now (cf. new screenshots in first post), @fregante you can remove my access.

Any other special cases this feature should handle? If not, I think it's done.

**Review by yakov116 — APPROVED** [review]
LGTM

**Comment by fregante** [comment]
🥳

**Comment by yakov116** [comment]
@cheap-glitch Can you take a look at https://github.com/xojs/xo/issues/338 its missing the label header.

Thanks

**Comment by cheap-glitch** [comment]
> its missing the label header

The "Label" heading is always hidden, except when the user can edit the sidebar. This PR doesn't change that

**Pull request merged**